File: patterns/java/src/refactoring_guru/patterns/visitor/example/shapes/Circle.java
package refactoring_guru.patterns.visitor.example.shapes;

import refactoring_guru.patterns.visitor.example.Visitor;

public class Circle extends Dot {
    public int radius;

    public Circle(int id, int x, int y, int radius) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.radius = radius;
    }

    @Override
    public String accept(Visitor visitor) {
        return visitor.visitCircle(this);
    }

    public int getRadius() {
        return radius;
    }
}

File: patterns/cs/src/RefactoringGuru/Visitor/Example/Shapes/Circle.cs
using System;
using RefactoingGuru.Visitor.Example.Visitor;

namespace RefactoingGuru.Visitor.Example.Shapes
{
    class Circle : Dot
    {
        public int radius;

        public Circle(int id, int x, int y, int radius)
        {
            this.id = id;
            this.x = x;
            this.y = y;
            this.radius = radius;
        }

        public String Accept(IVisitor visitor)
        {
            return visitor.VisitCircle(this);
        }

        public int GetRadius()
        {
            return radius;
        }
    }
}
